app/account/components/AccountDetails/components/AvatarUpdate/cropUtils.ts

```diff
@@ -0,0 +1,32 @@
+export const getCroppedImg = async (imageSrc: string, crop: { x: number; y: number; width: number; height: number }, userName: string) => {
+  const avatarMaxSizeInPx = 192
+  const image = new Image()
+  image.src = imageSrc
+
+  return new Promise<File>((resolve, reject) => {
+    image.onload = () => {
+      const canvas = document.createElement("canvas")
+      const ctx = canvas.getContext("2d")
+
+      if (!ctx) {
+        return reject(new Error("Failed to get canvas context"))
+      }
+
+      const size = 129
+
+      canvas.width = size
+      canvas.height = size
+
+      ctx.drawImage(image, crop.x, crop.y, crop.width, crop.height, 0, 0, canvas.width, canvas.height)
+
+      canvas.toBlob((blob) => {
+        if (!blob) {
+          return reject(new Error("Failed to create blob"))
+        }
+        resolve(new File([blob], `${userName.replaceAll(" ", "_")}_avatar`, { type: "image/jpeg" }))
+      }, "image/jpeg")
+    }
+
+    image.onerror = (error) => reject(error)
```

Comment:
Related to the error comments on the `AvatarUpdate` component, do you think this is sufficient enough for error handling or should we reject with and `Error` object? Again, might be overkill, but curious about your opinion on these.

Reply:
I feel this is good enough. The point for the users is that something failed and that they get to try again

Reply:
Sounds good, wanted to throw it out there.